curator-framework/src/test/java/org/apache/curator/framework/imps/TestWatcherIdentity.java

```diff
@@ -30,169 +36,141 @@
-        assertEquals(set.size(), 1);
-    }
-
-    @Test
-    public void testCuratorWatcher() throws Exception
-    {
-        Timing timing = new Timing();
-        CountCuratorWatcher watcher = new CountCuratorWatcher();
-        CuratorFramework client = CuratorFrameworkFactory.newClient(server.getConnectString(), timing.session(), timing.connection(), new RetryOneTime(1));
-        try
-        {
-            client.start();
-            client.create().forPath(PATH);
-            // Add twice the same watcher on the same path
-            client.getData().usingWatcher(watcher).forPath(PATH);
-            client.getData().usingWatcher(watcher).forPath(PATH);
-            // Ok, let's test it
-            client.setData().forPath(PATH, new byte[]{});
-            timing.sleepABit();
-            assertEquals(1, watcher.count.get());
-        }
-        finally
-        {
-            CloseableUtils.closeQuietly(client);
-        }
-    }
-
-
-    @Test
-    public void testZKWatcher() throws Exception
-    {
-        Timing timing = new Timing();
-        CountZKWatcher watcher = new CountZKWatcher();
-        CuratorFramework client = CuratorFrameworkFactory.newClient(server.getConnectString(), timing.session(), timing.connection(), new RetryOneTime(1));
-        try
-        {
-            client.start();
-            client.create().forPath(PATH);
-            // Add twice the same watcher on the same path
-            client.getData().usingWatcher(watcher).forPath(PATH);
-            client.getData().usingWatcher(watcher).forPath(PATH);
-            // Ok, let's test it
-            client.setData().forPath(PATH, new byte[]{});
-            timing.sleepABit();
-            assertEquals(1, watcher.count.get());
-        }
-        finally
-        {
-            CloseableUtils.closeQuietly(client);
-        }
-    }
+public class TestWatcherIdentity extends BaseClassForTests {
+	private static final String PATH = "/foo";
+	private static final int TIMEOUT_MS = 100000;
+
+	private static class CountZKWatcher implements Watcher {
+		private final AtomicInteger count = new AtomicInteger(0);
+
+		@Override
+		public void process(WatchedEvent event) {
+			count.incrementAndGet();
+		}
+	}
+
+	@Test
+	public void testSameWatcherPerZKDocs() throws Exception {
+		CountZKWatcher actualWatcher = new CountZKWatcher();
+		Timing timing = new Timing();
+		CuratorFramework client = CuratorFrameworkFactory.newClient(server.getConnectString(), timing.session(),
+				timing.connection(), new RetryOneTime(1));
+		try {
+			client.start();
+			client.create().forPath("/test");
+
+			// per ZK docs, this watcher should only trigger once
+			client.checkExists().usingWatcher(actualWatcher).forPath("/test");
+			client.getData().usingWatcher(actualWatcher).forPath("/test");
+
+			client.setData().forPath("/test", "foo".getBytes());
+			client.delete().forPath("/test");
+			Awaitility.waitAtMost(Duration.ofMillis(TIMEOUT_MS))
```

Comment:
We can use the default time out

Reply:
Changed `waitAtMost(Duration)` to `await()`